neuralcoref/train/conllparser.py

```diff
@@ -348,11 +348,11 @@ def add_conll_utterance(
         # Convert conll tokens coref index in spacy tokens indexes
         identified_gold = [False] * len(corefs)
         for coref in corefs:
-            assert (
-                coref["label"] is not None
-                and coref["start"] is not None
-                and coref["end"] is not None
-            ), ("Error in coreference " + coref + " in " + parsed)
+            missing_values = [key for key in ['label', 'start', 'end', ] if coref[key] is None]
+            if missing_values:
+                found_values = {key: coref[key] for key in ['label', 'start', 'end'] if coref[key] is not None}
+                raise Exception(f"Coref {str(found_values)} with empty field(s) {', '.join(missing_values)} found in {self.name}")
```

Comment:
You don't need to use `str()` in f strings. Also here I would rewrite `coref[key]` to `coref.get(key, None)` instead.

Then for readability, I would rewrite it a little, like so:
```
f"Coref {self.name} with fields {found_values} has empty vaues for the keys {missing_values}."
```

Reply:
```suggestion
                raise Exception(f"Coref {self.name} with fields {found_values} has empty values for the keys {missing_values}.")
```